test/test_utils_waitcounterfilelock.py

```diff
@@ -0,0 +1,16 @@
+# Owner(s): ["module: unknown"]
+import tempfile
+
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils.waitcounterfilelock import WaitCounterFileLock
+
+
+class TestWaitCounterFileLock(TestCase):
+    def test_checkpoint_trigger(self):
+        _, p = tempfile.mkstemp()
+        with WaitCounterFileLock(p):
+            pass
```

Comment:
What does this really test though? Is there a straightforward way to check whether it behaves like a file lock without, e.g., launching a shitty helper thread that competes for the same lock? Is there a non-blocking mode for example?

Reply:
We're primarily just checking for crashes. Let me add a test which makes sure we get some sequencing as a smoke test.

I'd really liek to check that the waitcounters are written, but we don't have the backend to test this :(.